beebop/filestore.py

```diff
@@ -136,13 +134,44 @@ def network_file(self, p_hash) -> str:
         """
         return str(PurePath(self.output(p_hash), f"{p_hash}_graph.gt"))
 
+    def external_previous_query_clustering_path(self, p_hash) -> str:
+        """
+        Generates the file path for the external
+        previous query clustering results.
+
+        :param p_hash (str): The hash value representing the query.
+
+        :return str: [The file path to the external]
+        previous query clustering CSV file.
+        """
+        return str(
+            PurePath(self.output(p_hash), f"{p_hash}_external_clusters.csv")
+        )
+
     def previous_query_clustering(self, p_hash) -> str:
         """
+        Returns previous query clustering csv file.
+
         :param p_hash: [project hash]
         :return str: [path to previous clustering file]
         """
-        return str(PurePath(self.output(p_hash),
-                            f"{p_hash}_external_clusters.csv"))
+        return (
+            self.external_previous_query_clustering_path(p_hash)
+            if self.has_external_previous_query_clustering(p_hash)
+            else str(PurePath(self.output(p_hash), f"{p_hash}_clusters.csv"))
```

Comment:
So this is all supporting species without external clustering as it just returns the internal clusters csv? But I'm a bit confused about the path through the code here - has_external_previous_clustering checks for the file's existence, but when does that get written? What does 'previous' mean here?